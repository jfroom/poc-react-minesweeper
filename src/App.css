.navbar.navbar-inverse {
  border-radius: 0;
  font-family: 'Roboto Slab', serif;
}
.navbar-header {
  float: left;
}
.navbar-right {
  float: right;
  margin-right: -15px;
}
.btn {
  margin: 5px;
}
.App {
  color: black;
}
.board {
  display: block;
  outline: 4px solid #666;
  margin: 20px auto;
  width: calc(8 * 32px);
  height: calc(8 * 32px);
  font-size: 0;
}

.tile-wrap {
  position: relative;
  top: 0;
  left: 0;
  display: inline-block;
  width: 32px;
  height: 32px;
}
.tile, .tile-cover {
  font-size: 22px;
  font-weight: bold;
  vertical-align: top;
  cursor: default;

  /* Tricky borders because want no deadspace between tiles which causes cursor to flicker between pointer/default */
  position: relative;
  top: -1px;
  left: -1px;
  border: 1px solid #666;
  width: calc(100% + 2px);
  height: calc(100% + 2px);
  background: white;
}
.tile-cover {
  position: absolute;
  background-color: #ccc;
  opacity: .95;
}

/* CSS glitch when empty div - so give it a space character */
.tile:empty:after, .tile-cover:empty:after {
  content: '\200b';
}

.tile .glyphicon {
  display: inline-block;
  position: relative;
  top: 3px;
}
.glyphicon-fire {
  color: brown;
}

.tile.clicked {
  background-color: red;
}

/* TODO: SCSS would clean this up! */
.App.is-active .tile-cover {
  cursor: pointer;
}
.App:not(.is-active).is-winner .tile-cover {
  opacity: .5;
}
.App:not(.is-active).is-winner .tile {
  background-color: green;
}
.App:not(.is-active):not(.is-winner) .tile-cover{
  opacity: .53
}
.App:not(.is-active):not(.is-winner) .tile {
  background-color: red;
}
