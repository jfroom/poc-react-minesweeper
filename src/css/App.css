.App {
  text-align: center;
}
.navbar.navbar-inverse {
  border-radius: 0;
  font-family: 'Roboto Slab', serif;
}
.btn {
  margin: 5px;
}
