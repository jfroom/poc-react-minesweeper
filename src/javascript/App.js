
import React, { Component } from 'react';
import { Navbar, Button } from 'react-bootstrap';
import '../css/App.css';

class App extends Component {
  render() {
    return (
      <div>
        <Navbar bsStyle='inverse'>
          <div className="container-fluid">
            <div className="navbar-header">
              <a className="navbar-brand" href="#">Minesweeper</a>
            </div>
          </div>
        </Navbar>
        <div className="App container-fluid">
          <HUD/>
        </div>
      </div>
    );
  }
}

class HUD extends Component {
  render() {
    return (
      <section role='hud'>
        <Button bsStyle="primary" bsSize="large">New Game</Button>
        <Button bsStyle="warning" bsSize="large">Validate</Button>
      </section>
    )
  }
}
export default App;
