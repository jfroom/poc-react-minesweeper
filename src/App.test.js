import React from 'react';
import {Tile, Board, HUD, App} from './App';
import toJson from 'enzyme-to-json';
import _ from 'underscore';

// TILE
describe ('Tile Component', () => {
  it ('empty & no cover', () => {
    const wrap = shallow(<Tile onClick={() => {}}/>);
    expect(toJson(wrap)).toMatchSnapshot();

    // Empty
    expect(wrap.find('.tile')).not.toHaveText();

    // Has no cover
    expect(wrap.find('.tile-cover')).toBeEmpty();
  });

  it ('covered with content & click event testing', () => {
    // Mock click handler
    const onClick = jest.fn();

    // Create shallow wrap & Assert snapshot
    const wrap = shallow(<Tile isCovered={true} dist={1} onClick={onClick}/>);
    expect(toJson(wrap)).toMatchSnapshot();

    // Empty
    expect(wrap.find('.tile')).toHaveText('1');

    // Has cover
    expect(wrap.find('.tile-cover')).toBePresent();

    // Simulate click
    wrap.find('.tile-cover').simulate('click');

    // Mock called once with no arguments
    expect(onClick.mock.calls.length).toBe(1);
  });

  it ('snapshot Tile, with mine', () => {
    const wrap = shallow(<Tile hasMine={true} />);
    expect(toJson(wrap)).toMatchSnapshot();

    // Mine
    expect(wrap.find('Glyphicon')).toBePresent();
  });

});

// BOARD
describe ('Board Component', () => {
  it('Board renders & handles tile clicks', () => {
    const tiles = [
      {isCovered: true, dist: 0},
      {isCovered: true, dist: 1},
      {isCovered: false, dist: 0},
      {isCovered: false, dist: 2}
    ];
    const onClickTile = jest.fn();
    const wrap = shallow(<Board tiles={tiles} onClickTile={onClickTile} />);
    expect(toJson(wrap)).toMatchSnapshot();
    expect(wrap.find('Tile').length).toBe(tiles.length);

    // Use mock to assert tile clicks
    const rTiles = wrap.find('Tile');
    expect(onClickTile.mock.calls.length).toBe(0);

    // Click first tile
    rTiles.at(0).simulate('click');
    expect(onClickTile.mock.calls.length).toBe(1);
    expect(onClickTile.mock.calls[0][0]).toBe(0);

    // Click last tile
    rTiles.at(tiles.length - 1).simulate('click');
    expect(onClickTile.mock.calls.length).toBe(2);
    expect(onClickTile.mock.calls[1][0]).toBe(tiles.length - 1);
  });
});

// HUD
describe ('HUD Component', () => {
  it('Game Active', () => {
    const onClickValidate = jest.fn();
    const onClickNewGame = jest.fn();
    const wrap = shallow(
      <HUD isGameActive={true} onClickValidate={onClickValidate} onClickNewGame={onClickNewGame}/>
    );
    expect(toJson(wrap)).toMatchSnapshot();

    wrap.find('#validate').simulate('click');
    expect(onClickValidate.mock.calls.length).toBe(1);
    expect(onClickNewGame.mock.calls.length).toBe(0);

    wrap.find('#newgame').simulate('click');
    expect(onClickValidate.mock.calls.length).toBe(1);
    expect(onClickNewGame.mock.calls.length).toBe(1);
  });
  it('Game over with winner', () => {
    const wrap = shallow(
      <HUD isGameActive={false} isWinner={true} onClickValidate={() => {}} onClickNewGame={() => {}}/>
    );
    expect(toJson(wrap)).toMatchSnapshot();
  });
  it('Game over with loser', () => {
    const wrap = shallow(
      <HUD isGameActive={false} isWinner={false} onClickValidate={() => {}} onClickNewGame={() => {}}/>
    );
    expect(toJson(wrap)).toMatchSnapshot();
  });
});

// APP
describe ('App Component', () => {
  it('Renders, loses, resets, wins', () => {
    const mineIndices =
      [0, 5, 6, 8, 20, 21, 31, 38, 45, 55];
    const tileDistances = [
      0,2,0,0,1,0,0,1,
      0,2,0,1,3,4,3,1,
      1,1,0,1,0,0,2,1,
      0,0,0,1,2,3,3,0,
      0,0,0,0,1,2,0,2,
      0,0,0,0,1,0,3,2,
      0,0,0,0,1,1,2,0,
      0,0,0,0,0,0,1,1
    ];

    // Set main App wrap & assert baseline snapshot
    const wrap = shallow(<App mineIndices={mineIndices} />);
    expect(toJson(wrap)).toMatchSnapshot();

    // Assert tile distances
    const distances = wrap.find('Board').prop('tiles').map(tile => tile.dist);
    expect(distances).toEqual(tileDistances);

    // All tiles covered
    const numTilesCovered = () =>
      _.countBy(
        wrap.find('Board').prop('tiles'),
        (tile) => tile.isCovered
      );
    const clickTile = (idx) => wrap.find('Board').prop('onClickTile')(idx);
    expect(numTilesCovered()).toEqual({"true": 64});

    // Game starts with default HUD
    expect(wrap).toHaveState('isActive', true);
    let hud = wrap.find('HUD');
    expect(hud).toHaveProp('isGameActive', true);

    // Clicking on mine loses game
    wrap.simulate('clickTile');
    clickTile(mineIndices[0]);
    hud = wrap.find('HUD');
    expect(wrap).toHaveState('isActive', false);
    expect(hud).toHaveProp('isGameActive', false);
    expect(hud).toHaveProp('isWinner', false);
    expect(wrap.find(".App")).not.toHaveClassName('is-winner');
    expect(numTilesCovered()).toEqual({true: 63, false: 1});
    expect(wrap.find('Board').props('tiles')).toMatchSnapshot();


    // Start new game
    wrap.find('HUD').prop('onClickNewGame')();
    expect(numTilesCovered()).toEqual({true: 64});

    // Uncover distance of '2'
    clickTile(1);
    expect(wrap).toHaveState('isActive', true);
    expect(numTilesCovered()).toEqual({true: 63, false: 1});

    // Chain uncover
    clickTile(2);
    expect(wrap).toHaveState('isActive', true);
    var test = {true: 23, false: 41};
    expect(numTilesCovered()).toEqual(test);

    // Uncover one at a time
    for (let idx of [7, 13, 14, 15, 22, 23, 29, 30, 37, 39, 46, 47, 63]) {
      clickTile(idx);
      test.true--;
      test.false++;
      expect(numTilesCovered()).toEqual(test);
    }

    // Winner!
    wrap.find('HUD').prop('onClickValidate')();
    expect(wrap).toHaveState('isActive', false);
    expect(wrap).toHaveState('isWinner', true);
    hud = wrap.find('HUD');
    expect(hud).toHaveProp('isGameActive', false);
    expect(hud).toHaveProp('isWinner', true);
  });
});
