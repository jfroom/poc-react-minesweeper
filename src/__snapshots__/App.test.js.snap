exports[`App Component Renders, loses, resets, wins 1`] = `
<div>
  <Header />
  <div
    className="App container text-center  is-active">
    <HUD
      isGameActive={true}
      isWinner={false}
      onClickNewGame={[Function]}
      onClickValidate={[Function]} />
    <Board
      onClickTile={[Function]}
      tiles={
        Array [
          Object {
            "dist": 0,
            "hasMine": true,
            "isCovered": true,
            "x": 0,
            "y": 0,
          },
          Object {
            "dist": 2,
            "hasMine": false,
            "isCovered": true,
            "x": 1,
            "y": 0,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 2,
            "y": 0,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 3,
            "y": 0,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 4,
            "y": 0,
          },
          Object {
            "dist": 0,
            "hasMine": true,
            "isCovered": true,
            "x": 5,
            "y": 0,
          },
          Object {
            "dist": 0,
            "hasMine": true,
            "isCovered": true,
            "x": 6,
            "y": 0,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 7,
            "y": 0,
          },
          Object {
            "dist": 0,
            "hasMine": true,
            "isCovered": true,
            "x": 0,
            "y": 1,
          },
          Object {
            "dist": 2,
            "hasMine": false,
            "isCovered": true,
            "x": 1,
            "y": 1,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 2,
            "y": 1,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 3,
            "y": 1,
          },
          Object {
            "dist": 3,
            "hasMine": false,
            "isCovered": true,
            "x": 4,
            "y": 1,
          },
          Object {
            "dist": 4,
            "hasMine": false,
            "isCovered": true,
            "x": 5,
            "y": 1,
          },
          Object {
            "dist": 3,
            "hasMine": false,
            "isCovered": true,
            "x": 6,
            "y": 1,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 7,
            "y": 1,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 0,
            "y": 2,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 1,
            "y": 2,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 2,
            "y": 2,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 3,
            "y": 2,
          },
          Object {
            "dist": 0,
            "hasMine": true,
            "isCovered": true,
            "x": 4,
            "y": 2,
          },
          Object {
            "dist": 0,
            "hasMine": true,
            "isCovered": true,
            "x": 5,
            "y": 2,
          },
          Object {
            "dist": 2,
            "hasMine": false,
            "isCovered": true,
            "x": 6,
            "y": 2,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 7,
            "y": 2,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 0,
            "y": 3,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 1,
            "y": 3,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 2,
            "y": 3,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 3,
            "y": 3,
          },
          Object {
            "dist": 2,
            "hasMine": false,
            "isCovered": true,
            "x": 4,
            "y": 3,
          },
          Object {
            "dist": 3,
            "hasMine": false,
            "isCovered": true,
            "x": 5,
            "y": 3,
          },
          Object {
            "dist": 3,
            "hasMine": false,
            "isCovered": true,
            "x": 6,
            "y": 3,
          },
          Object {
            "dist": 0,
            "hasMine": true,
            "isCovered": true,
            "x": 7,
            "y": 3,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 0,
            "y": 4,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 1,
            "y": 4,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 2,
            "y": 4,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 3,
            "y": 4,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 4,
            "y": 4,
          },
          Object {
            "dist": 2,
            "hasMine": false,
            "isCovered": true,
            "x": 5,
            "y": 4,
          },
          Object {
            "dist": 0,
            "hasMine": true,
            "isCovered": true,
            "x": 6,
            "y": 4,
          },
          Object {
            "dist": 2,
            "hasMine": false,
            "isCovered": true,
            "x": 7,
            "y": 4,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 0,
            "y": 5,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 1,
            "y": 5,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 2,
            "y": 5,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 3,
            "y": 5,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 4,
            "y": 5,
          },
          Object {
            "dist": 0,
            "hasMine": true,
            "isCovered": true,
            "x": 5,
            "y": 5,
          },
          Object {
            "dist": 3,
            "hasMine": false,
            "isCovered": true,
            "x": 6,
            "y": 5,
          },
          Object {
            "dist": 2,
            "hasMine": false,
            "isCovered": true,
            "x": 7,
            "y": 5,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 0,
            "y": 6,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 1,
            "y": 6,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 2,
            "y": 6,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 3,
            "y": 6,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 4,
            "y": 6,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 5,
            "y": 6,
          },
          Object {
            "dist": 2,
            "hasMine": false,
            "isCovered": true,
            "x": 6,
            "y": 6,
          },
          Object {
            "dist": 0,
            "hasMine": true,
            "isCovered": true,
            "x": 7,
            "y": 6,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 0,
            "y": 7,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 1,
            "y": 7,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 2,
            "y": 7,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 3,
            "y": 7,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 4,
            "y": 7,
          },
          Object {
            "dist": 0,
            "hasMine": false,
            "isCovered": true,
            "x": 5,
            "y": 7,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 6,
            "y": 7,
          },
          Object {
            "dist": 1,
            "hasMine": false,
            "isCovered": true,
            "x": 7,
            "y": 7,
          },
        ]
      } />
  </div>
</div>
`;

exports[`App Component Renders, loses, resets, wins 2`] = `
Object {
  "onClickTile": [Function],
  "tiles": Array [
    Object {
      "dist": 0,
      "hasMine": true,
      "isCovered": false,
      "x": 0,
      "y": 0,
    },
    Object {
      "dist": 2,
      "hasMine": false,
      "isCovered": true,
      "x": 1,
      "y": 0,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 2,
      "y": 0,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 3,
      "y": 0,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 4,
      "y": 0,
    },
    Object {
      "dist": 0,
      "hasMine": true,
      "isCovered": true,
      "x": 5,
      "y": 0,
    },
    Object {
      "dist": 0,
      "hasMine": true,
      "isCovered": true,
      "x": 6,
      "y": 0,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 7,
      "y": 0,
    },
    Object {
      "dist": 0,
      "hasMine": true,
      "isCovered": true,
      "x": 0,
      "y": 1,
    },
    Object {
      "dist": 2,
      "hasMine": false,
      "isCovered": true,
      "x": 1,
      "y": 1,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 2,
      "y": 1,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 3,
      "y": 1,
    },
    Object {
      "dist": 3,
      "hasMine": false,
      "isCovered": true,
      "x": 4,
      "y": 1,
    },
    Object {
      "dist": 4,
      "hasMine": false,
      "isCovered": true,
      "x": 5,
      "y": 1,
    },
    Object {
      "dist": 3,
      "hasMine": false,
      "isCovered": true,
      "x": 6,
      "y": 1,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 7,
      "y": 1,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 0,
      "y": 2,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 1,
      "y": 2,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 2,
      "y": 2,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 3,
      "y": 2,
    },
    Object {
      "dist": 0,
      "hasMine": true,
      "isCovered": true,
      "x": 4,
      "y": 2,
    },
    Object {
      "dist": 0,
      "hasMine": true,
      "isCovered": true,
      "x": 5,
      "y": 2,
    },
    Object {
      "dist": 2,
      "hasMine": false,
      "isCovered": true,
      "x": 6,
      "y": 2,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 7,
      "y": 2,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 0,
      "y": 3,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 1,
      "y": 3,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 2,
      "y": 3,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 3,
      "y": 3,
    },
    Object {
      "dist": 2,
      "hasMine": false,
      "isCovered": true,
      "x": 4,
      "y": 3,
    },
    Object {
      "dist": 3,
      "hasMine": false,
      "isCovered": true,
      "x": 5,
      "y": 3,
    },
    Object {
      "dist": 3,
      "hasMine": false,
      "isCovered": true,
      "x": 6,
      "y": 3,
    },
    Object {
      "dist": 0,
      "hasMine": true,
      "isCovered": true,
      "x": 7,
      "y": 3,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 0,
      "y": 4,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 1,
      "y": 4,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 2,
      "y": 4,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 3,
      "y": 4,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 4,
      "y": 4,
    },
    Object {
      "dist": 2,
      "hasMine": false,
      "isCovered": true,
      "x": 5,
      "y": 4,
    },
    Object {
      "dist": 0,
      "hasMine": true,
      "isCovered": true,
      "x": 6,
      "y": 4,
    },
    Object {
      "dist": 2,
      "hasMine": false,
      "isCovered": true,
      "x": 7,
      "y": 4,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 0,
      "y": 5,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 1,
      "y": 5,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 2,
      "y": 5,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 3,
      "y": 5,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 4,
      "y": 5,
    },
    Object {
      "dist": 0,
      "hasMine": true,
      "isCovered": true,
      "x": 5,
      "y": 5,
    },
    Object {
      "dist": 3,
      "hasMine": false,
      "isCovered": true,
      "x": 6,
      "y": 5,
    },
    Object {
      "dist": 2,
      "hasMine": false,
      "isCovered": true,
      "x": 7,
      "y": 5,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 0,
      "y": 6,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 1,
      "y": 6,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 2,
      "y": 6,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 3,
      "y": 6,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 4,
      "y": 6,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 5,
      "y": 6,
    },
    Object {
      "dist": 2,
      "hasMine": false,
      "isCovered": true,
      "x": 6,
      "y": 6,
    },
    Object {
      "dist": 0,
      "hasMine": true,
      "isCovered": true,
      "x": 7,
      "y": 6,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 0,
      "y": 7,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 1,
      "y": 7,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 2,
      "y": 7,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 3,
      "y": 7,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 4,
      "y": 7,
    },
    Object {
      "dist": 0,
      "hasMine": false,
      "isCovered": true,
      "x": 5,
      "y": 7,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 6,
      "y": 7,
    },
    Object {
      "dist": 1,
      "hasMine": false,
      "isCovered": true,
      "x": 7,
      "y": 7,
    },
  ],
}
`;

exports[`Board Component Board renders & handles tile clicks 1`] = `
<div
  className="board">
  <Tile
    dist={0}
    isCovered={true}
    onClick={[Function]} />
  <Tile
    dist={1}
    isCovered={true}
    onClick={[Function]} />
  <Tile
    dist={0}
    isCovered={false}
    onClick={[Function]} />
  <Tile
    dist={2}
    isCovered={false}
    onClick={[Function]} />
</div>
`;

exports[`HUD Component Game Active 1`] = `
<div
  className="hud">
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="large"
    bsStyle="primary"
    disabled={false}
    id="newgame"
    onClick={[Function]}>
    New Game
  </Button>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="large"
    bsStyle="warning"
    disabled={false}
    id="validate"
    onClick={[Function]}>
    Validate
  </Button>
</div>
`;

exports[`HUD Component Game over with loser 1`] = `
<div
  className="hud">
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="large"
    bsStyle="primary"
    disabled={false}
    id="newgame"
    onClick={[Function]}>
    New Game
  </Button>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="large"
    bsStyle="danger"
    disabled={true}
    id="result">
    Loser!
  </Button>
</div>
`;

exports[`HUD Component Game over with winner 1`] = `
<div
  className="hud">
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="large"
    bsStyle="primary"
    disabled={false}
    id="newgame"
    onClick={[Function]}>
    New Game
  </Button>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="large"
    bsStyle="success"
    disabled={true}
    id="result">
    Winner!
  </Button>
</div>
`;

exports[`Tile Component covered with content & click event testing 1`] = `
<div
  className="tile-wrap">
  <div
    className="tile">
    1
  </div>
  <div
    className="tile-cover"
    onClick={[Function]} />
</div>
`;

exports[`Tile Component empty & no cover 1`] = `
<div
  className="tile-wrap">
  <div
    className="tile" />
</div>
`;

exports[`Tile Component snapshot Tile, with mine 1`] = `
<div
  className="tile-wrap">
  <div
    className="tile">
    <Glyphicon
      bsClass="glyphicon"
      glyph="fire" />
  </div>
</div>
`;
